From: Christian Brauner <brauner@kernel.org>
Date: Mon, 29 Sep 2025 11:41:16 +0200
Subject: [PATCH] mount: handle NULL values in mnt_ns_release()
Git-commit: 6c7ca6a02f8f9549a438a08a23c6327580ecf3d6
Patch-mainline: v6.18-rc1
References: bsc#1254308

When calling in listmount() mnt_ns_release() may be passed a NULL
pointer. Handle that case gracefully.

Signed-off-by: Christian Brauner <brauner@kernel.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: David Sterba <dsterba@suse.com>
---
 fs/namespace.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/namespace.c
+++ b/fs/namespace.c
@@ -144,7 +144,7 @@ static void mnt_ns_release(struct mnt_na
 	lockdep_assert_not_held(&mnt_ns_tree_lock);
 
 	/* keep alive for {list,stat}mount() */
-	if (refcount_dec_and_test(&ns->passive)) {
+	if (ns && refcount_dec_and_test(&ns->passive)) {
 		put_user_ns(ns->user_ns);
 		kfree(ns);
 	}
